### PR TITLE
Fetch Athena QueryExecution for result location

### PR DESCRIPTION
**Pull request by nyergler**

### CODE REVIEW

File: go/rows.go [hunk 1]
```diff
@@ -125,6 +129,16 @@ func (r *Rows) ColumnTypeDatabaseTypeName(index int) string {
...
+	r.resultsOpened.Do(func() {
+		// download the results and open the resulting CSV for reading
+		err = r.openResults()
+	})
+	if err != nil {
+		r.reachedLastPage = true
+		return err
+	}
+
 	if r.reachedLastPage {
```
Comment (jpalawaga) [inline go/rows.go:142]: this if condition could be combined with the above.

File: go/rows.go [hunk 2]
```diff
@@ -175,8 +194,8 @@ func (r *Rows) openResults() error {
 		_, err = r.mgr.DownloadWithContext(r.ctx,
 			scratchFile,
 			&s3.GetObjectInput{
-				Bucket: aws.String(r.config.GetOutputBucketname()),
-				Key:    aws.String(r.config.GetQueryResultKey(r.queryID)),
+				Bucket: aws.String(resultLocation.Host),
+				Key:    aws.String(strings.TrimPrefix(resultLocation.Path, "/")),
```
Comment (jpalawaga) [inline go/rows.go:198]: jw, what's the difference between the new and old versions of these statements?

Comment (nyergler) [reply go/rows.go:198]: The old version uses where we _told_ Athena to write the results; the new version uses where Athena _did_ write the results. AFAICT these will almost always be the same, but not always.

File: go/rows.go (hunk 1 above)
Comment (nyergler) [reply go/rows.go:142]: It is possible for `reachedLastPage` to be set elsewhere (it's effectively `closed`) and `err` would be `nil` after the first call to `Next`. Not the clearest, but I think it does what we want.

**Pull request merged**